crates/sparrow-instructions/src/inst.rs

```diff
@@ -261,11 +262,12 @@ impl PartialEq for InstKind {
     }
 }
 
-use std::hash::Hash;
 impl Hash for InstKind {
```

Comment:
Should include discriminant first. This is the macro expansion:

```
impl core::hash::Hash for InstKind {
    fn hash<H: core::hash::Hasher>(&self, ra_expand_state: &mut H) {
        core::mem::discriminant(self).hash(ra_expand_state);
        match self {
            InstKind::Simple(f0) => {
                f0.hash(ra_expand_state);
            }
            InstKind::FieldRef => {}
            InstKind::Cast(f0) => {
                f0.hash(ra_expand_state);
            }
            InstKind::Record => {}
            InstKind::Udf(f0) => {
                f0.hash(ra_expand_state);
            }
        }
    }
}
```

Consider something like:
```
enum Either {
  A(i64),
  B(i64),
}
```

You would likely want `A(5)` to hash differently from `B(5)`. You do that by including the discriminant.